src/peer/peer_manager.cpp

```diff
@@ -169,9 +176,70 @@ void PeerManager::HandleMessage() {
             }
             auto msg_from = GetPeer(msg.first);
             if (msg_from && msg_from->IsVaild()) {
-                msg_from->ProcessMessage(msg.second);
+                switch (msg.second->GetType()) {
+                    case NetMessage::BLOCK: {
+                        auto* b   = dynamic_cast<Block*>(msg.second.release());
+                        b->source = Block::NETWORK;
```

Comment:
```suggestion
                        if (!conditon) {
                             continue;
                         }
```
to reduce the not-good-looking code of if-inside-if